libV2/schemaUtils.js

```diff
@@ -736,6 +736,71 @@ let QUERYPARAM = 'query',
     return schema;
   },
 
+  /** process and resolves types from nested schema structure **/
+
+  processSchema = (resolvedSchema, parentRequired = new Set()) => {
+    if (resolvedSchema.type === 'object' && resolvedSchema.properties) {
+      const schemaDetails = {
+          type: resolvedSchema.type || 'unknown',
+          properties: {},
+          required: []
+        },
+        requiredProperties = new Set(resolvedSchema.required || []);
+
+      for (let [key, prop] of Object.entries(resolvedSchema.properties)) {
+        const propertyDetails = {
+          type: prop.type || 'unknown',
+          deprecated: prop.deprecated || false,
+          enum: prop.enum || undefined,
+          minLength: prop.minLength || undefined,
+          maxLength: prop.maxLength || undefined,
+          minimum: prop.minimum || undefined,
+          maximum: prop.maximum || undefined,
+          pattern: prop.pattern || undefined,
+          example: prop.example || undefined,
+          description: prop.description || undefined,
+          format: prop.format || undefined
+        };
+
+        if (requiredProperties.has(key) || parentRequired.has(key)) {
+          schemaDetails.required.push(key);
+        }
+        if (prop.$ref) {
```

Comment:
Where are we dereferencing this ? I can see we're calling `processSchema` again in this case, but this function doesn't seems to handle `ref` anywhere else. Am I missing something here ?